### PR TITLE
Fix sub second `config.reload.interval`

### PR DESCRIPTION
**Pull request by robbavey**
Due to a change in #11803, the sleep interval waiting for converge to
occur in this spec was effectively set to 0, as the sleep method was
using `to_seconds` on a `config.reload.interval` value set to `10ms` in
the spec. This change uses the `to_nanos` method to keep the functionality
as it was prior to this change.

### CODE REVIEW

**Comment by robbavey** [comment]
jenkins test this please

**Comment by robbavey** [comment]
jenkins test this please

File: logstash-core/spec/logstash/agent/converge_spec.rb [hunk 1]
```diff
@@ -191,7 +191,7 @@
           end
 
           it "it will keep trying to converge" do
-            sleep(agent_settings.get("config.reload.interval").to_seconds * 20) # let the interval reload a few times
+            sleep(agent_settings.get("config.reload.interval").to_nanos * 20 / 1_000_000_000.0) # let the interval reload a few times
```
Comment (jsvd) [inline logstash-core/spec/logstash/agent/converge_spec.rb:194]:
```suggestion
            # we can't do .to_seconds here as values under 1 seconds are rounded to 0
            # causing a race condition in the test. So we get the nanos and convert to seconds
            sleep(agent_settings.get("config.reload.interval").to_nanos * 1e-9 * 20) # let the interval reload a few times
```

Comment (robbavey) [reply logstash-core/spec/logstash/agent/converge_spec.rb]: I've made this change, and added a similar fix to `agent.rb`

File: logstash-core/lib/logstash/agent.rb [hunk 2]
```diff
@@ -85,7 +85,9 @@ def initialize(settings = LogStash::SETTINGS, source_loader = nil)
     end
 
     # Normalize time interval to seconds
-    @reload_interval = setting("config.reload.interval").to_seconds
+    # we can't do .to_seconds here as values under 1 seconds are rounded to 0
+    # causing a race condition in the test. So we get the nanos and convert to seconds
```
Comment (jsvd) [inline logstash-core/lib/logstash/agent.rb:89]:
```suggestion
    # so we get the nanos and convert to seconds instead.
```

Comment (robbavey) [reply logstash-core/lib/logstash/agent.rb]: 🤦 copy/paste ftw

**Review by jsvd — APPROVED** [review]
LGTM

**Comment by robbavey** [comment]
jenkins test this please

**Comment by jsvd** [comment]
@robbavey please backport this to 7.11, as builds are failing there as well (e.g. https://logstash-ci.elastic.co/job/elastic+logstash+7.11+multijob--ruby-unit-tests/11/console)

**Comment by robbavey** [comment]
@jsvd https://github.com/elastic/logstash/pull/12637

**Pull request merged**